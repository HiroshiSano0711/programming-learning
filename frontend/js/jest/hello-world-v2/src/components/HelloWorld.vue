<template>
  <div class="hello">
    <Message :message="msg" />
  </div>
</template>

<script>
import Message from './Message'

export default {
  name: 'HelloWorld',
  props: {
    msg: String
  },
  components: {
    Message
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
