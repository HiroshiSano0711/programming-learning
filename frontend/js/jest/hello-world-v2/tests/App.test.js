import { shallowMount } from '@vue/test-utils'
import HelloWorld from '../src/components/HelloWorld'

describe('HelloWorldコンポーネントのテスト', ()=>{
  it('propsに正常に値が渡っていること', ()=>{
    let wrapper = shallowMount(HelloWorld, {
      propsData: { msg: 'abc' }
    })
    expect(wrapper.props().msg).toEqual('abc')
  });

  it("期待されるHTML構造であること", () => {
    let wrapper = shallowMount(HelloWorld, {
      propsData: { msg: 'abc' }
    })
    expect(wrapper.element).toMatchSnapshot();
  });
});
