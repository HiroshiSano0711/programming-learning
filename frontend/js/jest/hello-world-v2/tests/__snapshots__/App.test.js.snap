// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorldコンポーネントのテスト 期待されるHTML構造であること 1`] = `
<div
  class="hello"
>
  <message-stub
    message="abc"
  />
</div>
`;
